### ISSUE TITLE
Clamp speed for multiway

### ISSUE THREAD

**Issue by matthijsgroen** [craftyjs/Crafty#1047]
I'd like to add a 'clamp' mechanic to the multiway.

It's effect: 

``` javascript
  directions = {UP_ARROW: -90, DOWN_ARROW: 90, RIGHT_ARROW: 0, LEFT_ARROW: 180};

  clamped = Crafty.e('2D, Multiway').multiway({x: 10, y: 10 }, directions, { clamped: true })
  current = Crafty.e('2D, Multiway').multiway({x: 10, y: 10 }, directions)

```

When the up arrow is pressed:

``` javascript
  clamped.vx // 0
  clamped.vy // -10

  current.vy // 0
  current.vy // -10
```

When both the up and the right arrow are pressed:

``` javascript
  clamped.vx // Math.sin(Math.PI / 4) (45deg) * 10 = 7.07
  clamped.vy // Math.cos(Math.PI / 4) (45deg) * -10 = -7.07

  current.vx // 10
  current.vy // -10
```

Should this become a new component like 'MultiwayClamped' or an option of the existing multiway?

**Comment by matthijsgroen** [comment]
Reason for me to want this feature is feedback I got on my game here: http://www.shmup-dev.com/forum/viewtopic.php?f=4&t=515&p=1563#p1562

**Comment by matthijsgroen** [comment]
See #1048 for details

**Comment by starwed** [comment]
Fixed (I think) by #1051, when `multiway` is called with `{normalize: true}`.
